controllers/controllers/services/cfservicebinding_controller.go

```diff
@@ -164,6 +173,76 @@ func (r *CFServiceBindingReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		}
 	}
 
+	if cfApp.Status.VCAPServicesSecretName == "" {
+		r.Log.Info("Did not find VCAPServiceSecret name on status of CFApp", "CFServiceBinding", cfServiceBinding.Name)
+		meta.SetStatusCondition(&cfServiceBinding.Status.Conditions, metav1.Condition{
+			Type:    VCAPServicesSecretAvailableCondition,
+			Status:  metav1.ConditionFalse,
+			Reason:  "SecretNotFound",
+			Message: "VCAPServicesSecret name absent from status of CFApp",
+		})
+
+		err = r.setStatus(ctx, cfServiceBinding)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
+	}
+
+	vcapServicesData, err := r.Builder.BuildVCAPServicesEnvValue(ctx, cfApp)
+	if err != nil {
+		r.Log.Error(err, "failed to build vcap services secret", "CFServiceBinding", cfServiceBinding)
+	}
+
+	vcapServicesSecret := new(corev1.Secret)
+	// Note: is there a reason to fetch the secret name from the service instance spec?
+	err = r.Client.Get(ctx, types.NamespacedName{Name: cfApp.Status.VCAPServicesSecretName, Namespace: req.Namespace}, vcapServicesSecret)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Consider a follow up story to avoid this logical branch by creating and populating the secret when absent
+			meta.SetStatusCondition(&cfServiceBinding.Status.Conditions, metav1.Condition{
+				Type:    VCAPServicesSecretAvailableCondition,
+				Status:  metav1.ConditionFalse,
+				Reason:  "SecretNotFound",
+				Message: "Secret does not exist",
+			})
+		} else {
+			// Current status describes the presence of the service instance secret
```

Comment:
How is this comment related to setting `false` to the `VCAPServicesSecretAvailableCondition` condition?